gantry/views.py

```diff
@@ -36,3 +37,83 @@ async def collect_job(request: web.Request) -> web.Response:
     )
 
     return web.Response(status=200)
+
+
+@routes.post("/v1/allocate")
```

Comment:
```suggestion
@routes.get("/v1/allocation")
```

The endpoint name `allocate` gives me the impression that we're allocating the resources in this request. Not a hard request by any means, but I might suggest changing it instead to `allocation` showing that folks are retrieving an allocation for a package. 

I'd also suggest accepting a `GET` request here instead of `POST`. Typically I've implemented/seen post requests reserved for actively modifying a database or system whereas this endpoint is non-destructive/retrieving data from the system.

Reply:
I was thinking [the same](https://github.com/spack/spack/pull/41622#discussion_r1470591283) 🤪

To make this work, we'd need to instruct the client to pass url-encoded JSON into one of the parameters. 

Here's an example request:

```
{"hash":"feifneifenife","package":{"name":"salmon","version":"1.10.2"},"compiler"{"name":"gcc","version":"7.3.1"}}
```

When encoded, it's around 200 bytes. The current restriction in aiohttp is 8190 bytes. I could increase the limit to something like 100k bytes (or larger, I'm not sure what the upper limit on the amount of specs we'd need to handle here is).

Reply:
@kwryankrattiger do you have an idea of how many specs would get sent to the API at once? I'm assuming it would be the size of the largest stack? 

Trying to think if there is some sort of limit we should have on amount of specs we'll process in a single request.

Reply:
We can make it one spec per request, that would make GET requests make more sense. GET `v1/dyanmic-mapping?data=<base64encodedjson>`

I would suspect we would be seeing up to a thousand specs per stack for the existing stacks. We are also working on a new internal workflow that prunes before we apply the configs which would reduce that to closer to 10s or 100s on average, but the max would still be high for the rebuild everything case.

Reply:
I'll plan for up to a thousand in a single request, and see how that goes

My instinct is that it makes sense to process as many specs as we can in a single request, as we can eliminate the overhead of thousands of requests at the same time. I don't know if a large request would wipe out these theoretical benefits though

Reply:
we can play with it, that is why I have the two modes because I wasn't sure if batching or individual requests would be better.

We can start with batch and see how it goes., we will be around a few MB for the larger messages, but if that is too large we can also look at setting a limit on the specs per requests.